### DISCUSSION TITLE
compiling help

### DISCUSSION THREAD

**Discussion by rden757** [Jyers/Marlin#1243] [Q&A] [answered]
Hello,
Tried several  times now, to compile and load Jyers version with higher temp settings for nylon.
And an all metal hot end.  This is for an ender 3 v2.  All I want to do is set higher temp limits.

So far, I have been trying to compile and build the Jyers files as given in the download.  Jyers 1.3.5
The PIO builds the bin file successfully.  
I get the .bin file at .pio/build/STM32F103RET6_creality/firmware-20210730-182057.bin.
But, but, but, this file does not load on to the 3D printer.  ender 3 v2

I think I'm missing something minor, but I could be way off, also.
Help would be greatly appreciated.

**Comment by rden757** [comment, marked as answer]
BTW, I have been launching VScode, and opening the Jyers 1.3.5 folder and hitting the check-mark.  And I have tried copying and replacing the confi.h and .adv.h with/from the configuration folder.  Perhaps something isn't hooked up correctly or completely, 
but it seems to compile and build a bin file each time.

**Reply by Cal-LF** [reply]
If you are getting a successful compile, you are mostly there.
Does it compile and load if no changes are made after copying the config files to the Marlin folder?
Are you using the Jyres default branch when you download the Zip file?
Is the SD card working otherwise? Able to flash back to previous FW? The filename must be different each time to flash.

**Reply by rden757** [reply]
I haven't made any changes yet to the config files.  So, it always compiles, but never loads onto the printer.
And, 
very happily, a pre-configured .bin, always loads and fixes everything.
===
Are you using the Jyres default branch when you download the Zip file? --- not sure about this

I have been going to the Jyers "releases"  at  ~/Jyers/Marlin/releases  and 
downloading the Source code (zip) at the bottom of the list.  and then opening this folder 
into VScode, along with Auto Build Marlin and PlatformIO, and hitting the check mark.  
It compiles, builds, I get the .bin file under .pio/build/STM32F103RET6 --- and this doesn't load on the printer.
I delete this file from the micro sd card, and load a pre-built .bin, which loads and works fine.

I've also tried to copy and replace the config.h and config.adv.h files from the Confiuration Files folder, 
into the Marlin folder, in the downloaded, Source Code (zip) folder.  --check mark compiles ok

I maybe have downloaded the wrong source.  I will look for a default fork.  I'm not yet sure about the fork stuff.
and not sure if the .bin file needs to be renamed.  the filename is always firmware-date-time --- 
the time is always different, so perhaps you need to change this -- in some way?


I really appreciate your help.

**Reply by Cal-LF** [reply]
@rden757 Are you using the E3V2 Template configs? PlatformIO.ini make sure the environment is STM32F103RET6_creality.

I just finished downloading the Source Code zip and copied the Manual Mesh 3X3 configs to the Marlin folder. Changed the board to V427 for my mainboard. No other changes made and it compiles and loads to the printer.
If you want the latest changes to Jyres after any releases, download the source zip from the green Code button.
[Compiling Steps.txt](https://github.com/Jyers/Marlin/files/6911219/Compiling.Steps.txt)

**Reply by rden757** [reply]
Using your 'steps' - I think I'm doing these things.
Let me ask this before I begin again with my trial and error, random monkey walk.
Just doing steps 1 thru 7, picking an E3V2 config file set, copying into Marlin, 
and check-mark compile and build, ----SHOULD it not load on to the printer 
with no changes?  
I have the 4.2.2 board.
I've set the envrs as you describe, in .pio ini
-much obliged

**Reply by Cal-LF** [reply]
@rden757 Yes it should.

**Reply by rden757** [reply]
acknowledged
